Sources/_MatchingEngine/Regex/Parse/LexicalAnalysis.swift

```diff
@@ -379,6 +375,43 @@ extension Source {
           "Unknown group kind '(?\(src.peek()!)'")
       }
 
+      // Explicitly spelled out PRCE2 syntax for some groups.
+      if src.tryEat("*") {
+        if src.tryEat(sequence: "atomic:") { return .atomicNonCapturing }
+
+        if src.tryEat(sequence: "pla:") ||
+            src.tryEat(sequence: "positive_lookahead:") {
+          return .lookahead
+        }
+        if src.tryEat(sequence: "nla:") ||
+            src.tryEat(sequence: "negative_lookahead:") {
+          return .negativeLookahead
+        }
+        if src.tryEat(sequence: "plb:") ||
+            src.tryEat(sequence: "positive_lookbehind:") {
+          return .lookbehind
+        }
+        if src.tryEat(sequence: "nlb:") ||
+            src.tryEat(sequence: "negative_lookbehind:") {
+          return .negativeLookbehind
+        }
+        if src.tryEat(sequence: "napla:") ||
+            src.tryEat(sequence: "non_atomic_positive_lookahead:") {
+          return .nonAtomicLookahead
+        }
+        if src.tryEat(sequence: "naplb:") ||
+            src.tryEat(sequence: "non_atomic_positive_lookbehind:") {
+          return .nonAtomicLookbehind
+        }
+        if src.tryEat(sequence: "sr:") || src.tryEat(sequence: "script_run:") {
+          return .scriptRun
+        }
+        if src.tryEat(sequence: "asr:") ||
+            src.tryEat(sequence: "atomic_script_run:") {
+          return .atomicScriptRun
+        }
+      }
+
```

Comment:
In the future, I think this code will be an example of composing consumers into ordered choices.

Reply:
CC @natecook1000 . This could be similar to combining regexes into grammars of syntax highlighters.